File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.cpp
class Solution {
public:
    bool isPalindrome(string s, int l, int r){
        while(l < r){
            if(s[l] == s[r]){
                l++;
                r--;
            }
            else return false;
        }
        return true;
    }
    bool validPalindrome(string s) {
        int n = s.size();
        int l = 0, r = n-1;
        while(l < r){
            if(s[l] != s[r]){
                return (isPalindrome(s, l,r-1) || isPalindrome(s, l+1, r));
            }
            l++;
            r--;
        }
        
       return true; 
    }
};

File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.java
class Solution {
    boolean isValid(String s, int l, int r){
        while(l < r){
            if(s.charAt(l) == s.charAt(r)){
                l++;
                r--;
            }
            else return false;
        }
        return true;
    }
    public boolean validPalindrome(String s) {
        int l = 0;
        int r = s.length() -1;
        while(l < r){
            if(s.charAt(l) != s.charAt(r)){
                return (isValid(s,l,r-1) || isValid(s,l+1,r));
            }
            l++;
            r--;
        }
        return true;
    }
}